<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>salt_and_pepper_Dialog</class>
 <widget class="QWidget" name="salt_and_pepper_Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>292</width>
    <height>154</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Настройки атаки &quot;Соль &amp; Перец&quot;</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/icons/salt_and_pepper.svg</normaloff>:/icons/icons/salt_and_pepper.svg</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <widget class="QLabel" name="label_level">
     <property name="text">
      <string>Уровень зашумления, %</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QSlider" name="amount_Slider">
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="sliderPosition">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksBelow</enum>
     </property>
     <property name="tickInterval">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QSpinBox" name="amount_spinBox">
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QLabel" name="label_salt_vs_pepper">
     <property name="text">
      <string>Соль vs Перец, %</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QSpinBox" name="salt_spinBox">
     <property name="toolTip">
      <string>Процент соли к перцу от общего уровня зашумления</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QSlider" name="salt_vs_pepper_Slider">
     <property name="minimum">
      <number>-50</number>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="sliderPosition">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksBelow</enum>
     </property>
     <property name="tickInterval">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QSpinBox" name="pepper_spinBox">
     <property name="toolTip">
      <string>Процент перца к соли от общего уровня зашумления</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox_dialog">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>amount_spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>amount_Slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>129</x>
     <y>97</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>amount_Slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>amount_spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>129</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox_dialog</sender>
   <signal>rejected()</signal>
   <receiver>salt_and_pepper_Dialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>221</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox_dialog</sender>
   <signal>accepted()</signal>
   <receiver>salt_and_pepper_Dialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>154</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
